src/Microsoft.AspNetCore.Mvc.Core/Internal/ViewEnginePath.cs

```diff
@@ -11,9 +11,26 @@ namespace Microsoft.AspNetCore.Mvc.Internal
 {
     public static class ViewEnginePath
```

Comment:
Class looks good.

One alternative approach would be to loop through what the `StringTokenizer` returns up front, checking for any special segments. But, I'm not sure :100:% fidelity between the check and the operation is required; you've excluded most of the cases where `ResolvePath(...)` does nothing useful.

Reply:
Yeah, that doesn't too bad.

Reply:
Thanks, like it.